src/os/bluestore/BlueStore.cc

```diff
@@ -2258,7 +2258,8 @@ bool BlueStore::ExtentMap::do_write_check_depth(
 
   bool do_collect = true;
   if (depth < g_conf->bluestore_gc_max_blob_depth) {
-    *blob_depth = 1 + depth;
+    if (head_overlap || tail_overlap)
```

Comment:
no much sense in this patch - when head_overlap ==false && tail_overlap==false depth is equal to 0 and we just performed a no-op by reassigning blob_depth to 1